hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/EntityCollectionPart.java

```diff
@@ -270,16 +275,35 @@ public TableGroupJoin createTableGroupJoin(
 			SqlAliasBaseGenerator aliasBaseGenerator,
 			SqlExpressionResolver sqlExpressionResolver,
 			SqlAstCreationContext creationContext) {
-		return collectionDescriptor.getAttributeMapping().createTableGroupJoin(
-				navigablePath,
-				lhs,
-				explicitSourceAlias,
-				sqlAstJoinType,
-				fetched,
-				aliasBaseGenerator,
-				sqlExpressionResolver,
-				creationContext
-		);
+		if ( collectionDescriptor.isInverse() ) {
```

Comment:
I don't think `isInverse` is the proper flag for this. IMO this should be controlled by the FK descriptor directionality. AFAIU the `isInverse` method just signals if collection changes should be flushed to the database, which might by coincidence be true/false in the same cases when the FK is _inverse_, but it not necessarily correct.